Sync/wallet to develop

Alexandr163r opened the pull request.

Commit:
-Добавил структуру сервиса Wallet.
-В Wallet.Domain добавил основные сущности и энамы.
-В Wallet.Infrastructure DbContext и первая миграция.

Commit:
-Добавлены контролеры для категорий
-Название в бд добавлена условие уникальности название категории
-Добавлен маппинг

Commit:
Обновил миграцию, поле name в категориях уникальное.

Commit:
Добавлена миграция для сохранения даты создания при апдейте

Commit:
Добавлены круд операции для акаунтов (кошельков)

Commit:
Забыл миграцию)

Commit:
Убрал вызов репозитории в контроллер, заменил на вызов сервиса

Commit:
Добавлены круд операции для тегов, с поиском по строке.

Commit:
Разбил свагер по группам, переименовал поле в таблице транзакций

Commit:
Доработан базовый функционал, добавлен unit of work для атомарности транзакций и изменения баланса счета

Commit:
потерял

Commit:
замерджил wallet в develop

Commit:
Убрал дубли из sln

Dovmial commented on Wallet.API/Controllers/Tags/SearchTagsByPrefixController.cs:
prefix зачем-то проверяется дважды. 
для limit не учтены случаи null и <=0

Alexandr163r replied:
Поправил, времнно в любом случае, хотел написать нормальные сервисы валидации для всех эндпоинтов.

Dovmial commented on Wallet.API/Program.cs:
builder.AddEnvironmentVariables(); дважды, это норм?

Alexandr163r replied:
Мой косяк убрал.

Dovmial commented on Wallet.API/appsettings.json:
так в гит не стоит отправлять, для этого переменные окружения и используются. В нашем то еще ладно.

Alexandr163r replied:
Если не против пока оставлю так, да и перед ретро переделывать не хотелось бы, но ты прав в конечном итоге все настройки в окружение перенести надо.

Commit:
Убрал лишней билдер, Добавил проверку на лимит

Commit:
Добавлен запуск миграции при старте проекта

Commit:
Удалены лишние using из program

Dovmial commented on Wallet.Application/Services/TransactionService.cs:
потом может все-таки enum для типа операции?

Alexandr163r replied:
Поправил

Dovmial commented on Wallet.Application/Wallet.Application.csproj:
этот слой не должен зависеть от инфраструктурного слоя

Dovmial commented on Wallet.Infrastructure/Data/DesignTimeDbContextFactory.cs:
подключение файла к приложению явно должно быть не тут

Dovmial commented on Wallet.Infrastructure/Data/Repositories/AccountRepository.cs:
AsNoTracking() для гет запросов. во всех репозиториях

Alexandr163r replied:
Добавил

Dovmial commented on Wallet.Infrastructure/Wallet.Infrastructure.csproj:
не зависит от Application, а должен.

Dovmial reviewed (approved):
поток управления нарушен, но это в рефакторинги будущие.

dariapritchina commented on Wallet.API/Controllers/Accounts/AccountBase.cs:
```diff
@@ -0,0 +1,10 @@
+
+[ApiController]
+[Route("api/accounts")]
+[ApiExplorerSettings(GroupName = "Управление аккаунтами")]
```
Может переименовать в "Управление счетами"?

Alexandr163r replied:
Поправил

Alexandr163r replied:
поменял

dariapritchina commented on Wallet.API/Controllers/Categories/CreateCategory.cs:
```diff
@@ -0,0 +1,27 @@
+public class CreateCategory : CategoryBase
+{
+    private readonly ICategoryService _categoryService;
+    private readonly IMapper _mapper;
```
Может categoryService и mapper вынести в базовый класс CategoryBase? На твое усмотрение, не критично :)

Alexandr163r replied:
Вынес

dariapritchina commented on Wallet.API/Models/Accounts/AccountResponse.cs:
```diff
@@ -0,0 +1,14 @@
+    public Guid UserId { get; set; }
+    public string Name { get; set; } = string.Empty;
+    public decimal CurrentBalance { get; set; }
+    public Currency Currency { get; set; }
```
Может тут по умолчанию сделать "RUB". Типа public Currency Currency { get; set; } = Currency.Rub;

dariapritchina commented on Wallet.Application/Services/TransactionService.cs:
```diff
@@ -0,0 +1,232 @@
+        account.CurrentBalance += existingTransaction.Amount;
+    }
+    
+    if (transactionDto.OperationType == 1) 
```
Может можно operationType 1,2 вынести в enum

Alexandr163r replied:
Да поправил, пока тестил что то как то забылся и накидал просто числами

Commit:
- Изменено название эндпоинтов на "Управление счетами"
- Добавлено значение валюты по умолчанию для AccountRequest
- Числовые значения заменены на OperationType

Commit:
- Вынес зависимости в базовые контролеры

Commit:
Добавлены .AsNoTracking() для всех гет запросов
